### DISCUSSION TITLE
How could I have multiple consumers listening to the same queue?

### DISCUSSION THREAD

**Discussion by alejandro-jara** [Havret/dotnet-activemq-artemis-client#410] [Q&A] [answered]
Could it be parameterized that an instance of a solution generates multiple consumers for the same queue? This in order to parallelize the work

**Comment by Havret** [comment, marked as answer]
Hello @alejandro-jara,

If you're using ArtemisNetClient directly, all you need to do is to spin up multiple consumers. If you're using `Extensions.DependencyInjection` package, it's even simpler. It's just a matter of setting `ConcurrentConsumers` property to define the desired level of concurrency:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddActiveMq(name: "my-artemis-cluster", endpoints: new[] { Endpoint.Create(host: "localhost", port: 5672, "artemis", "artemis") })
        .AddConsumer(address: "a1", RoutingType.Multicast, queue: "q2", new ConsumerOptions { ConcurrentConsumers = 5 }, async (message, consumer, _, _) =>
            {
                Console.WriteLine("q2: " + message.GetBody<string>());
                await consumer.AcceptAsync(message);
            })
            .EnableQueueDeclaration()
            .EnableAddressDeclaration();

    services.AddActiveMqHostedService();
}
```

**Reply by alejandro-jara** [reply]
Thank you very much for the quick response, it was exactly what I needed.
